internal/controller/kubeconfig/gardener_cluster_controller_test.go

```diff
@@ -68,6 +68,34 @@ var _ = Describe("Gardener Cluster controller", func() {
 			expectKubeconfigMetricsAreValid(metricsData, lastSyncTime, "Kubeconfig expiration metrics should be appended after creation", shootName)
 		})
 
+		//
+		It("Should handle gardener timeout", func() {
```

Comment:
This test doesn't exercise the actual fix.
It test whether an error returned from the `Provider.Fetch` is recorded in CR status. 

The main problems:
- `Provider.Fetch` is mocked so we bypass timeout check that happens inside the Gardener Client
- _Should handle gardener timeout_  and are very similar _Should set Error status on CR if failed to fetch kubeconfig_

The correct test should pass the cancelled context to the call to the Gardener Client. The problem is that we probably cannot achieve it with the current solution as we would need to establish connection to the Gardener cluster. Using any kind of fake is not an option here because we need to verify the behaviour of the real client.

What I propose is to get rid of _Should handle gardener timeout_, and modify _Should set Error status on CR if failed to fetch kubeconfig_ to check if error message returned from `Provider.Fetch` will be recorded in status.

Reply:
@akgalwas did you mean [this](https://github.com/kyma-project/infrastructure-manager/pull/271/files/2bb3e108502d7861d84bd5c5e86667b813d4aa02..eb9672ea400667e26612b9920a3a547151924e10)?